pkgs/by-name/ca/cargo-shear/package.nix

```
@@ -0,0 +1,24 @@
+{ fetchFromGitHub, lib, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-shear";
+  version = "0.0.24";
+
+  src = fetchFromGitHub {
+    owner = "Boshen";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-7eBFDmeFOXqZhNE4RDlIq991xEoCGL6XLMmUQuLJvrE=";
+  };
+
+  cargoSha256 = "sha256-DQjLW6NtUPERT82zIlrGRk3FffU/EfLBMgAeFfFdDxE=";
+
+  meta = with lib; {
+    description = "Detect and remove unused dependencies from Cargo.toml";
+    mainProgram = "cargo-shear";
+    homepage = "https://github.com/Boshen/cargo-shear";
+    changelog = "https://github.com/Boshen/cargo-shear/blob/v${version}/CHANGELOG.md";
```

```suggestion
    changelog = "https://github.com/Boshen/cargo-shear/blob/${src.rev}/CHANGELOG.md";
```

using src.rev here instead will make sure the changelog url doesn't break if someone updates this package in the future to use the `-unstable-YYYY-MM-DD` version format, and specifies a specific commit instead of a tag.